Fix Tesla update showing scheduled updates as installing

ptarjan opened the pull request:
When a Tesla software update is scheduled for later (e.g., tonight), the update entity was incorrectly showing as "Installing" because the SCHEDULED status was being treated the same as INSTALLING.

This change:
1. Makes in_progress only true when the status is INSTALLING
2. Adds an update_status attribute to show the actual API status (available, scheduled, installing, downloading, etc.)
3. Shows "Update scheduled" in the release_summary when scheduled, making it visible in the update card UI

Users can now clearly see when an update is scheduled vs installing.

In my current HA I see this when I have scheduled the update:

<img width="555" height="278" alt="image" src="https://github.com/user-attachments/assets/67b34926-547c-4db7-af53-32e9ba4cc4c8" />


<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [ ] I understand the code I am submitting and can explain how it works.
- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.
- [ ] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

ptarjan committed:
Fix Tesla update showing scheduled updates as installing

When a Tesla software update is scheduled for later (e.g., tonight),
the update entity was incorrectly showing as "Installing" because
the SCHEDULED status was being treated the same as INSTALLING.

This change:
1. Makes in_progress only true when the status is INSTALLING
2. Adds an update_status attribute to show the actual API status
   (available, scheduled, installing, downloading, etc.)
3. Shows "Update scheduled" in the release_summary when scheduled,
   making it visible in the update card UI

Users can now clearly see when an update is scheduled vs installing.

Bre77 commented:
@ptarjan can you tell me which you would prefer:
- You click install in Home Assistant or the Tesla app or schedule the install **nothing** happens for a minute (or more) until the install actually starts
- You schedule the install, and Home Assistant shows it as in progress with a percentage of 0

I think the first one is a bad UX, which is why it works how it works now. The 10% is what Tesla reports, even when scheduled, but I can force that to zero until it starts.

joostlek commented:
What's the status on this PR? If this PR is not the direction it's going to go in, I'd rather just close it

ptarjan committed:
Fix Tesla update showing scheduled updates as installing

When a Tesla software update is scheduled for later (e.g., tonight),
the update entity was incorrectly showing as "Installing" because
the SCHEDULED status was being treated the same as INSTALLING.

Now only shows as installing if:
- Status is INSTALLING (actually installing), or
- Status is SCHEDULED and scheduled_time_ms is within 2 minutes

This allows users to schedule updates for later without the UI
showing them as currently installing.

ptarjan commented:
@Bre77 I vibe coded what we talked about and pushed it here but I'm happy if you want to do it too

ptarjan committed:
Hide install button when Tesla update is already scheduled

When an update is already scheduled, hide the install button since
the user has already scheduled it. The button only shows when the
status is "available" (not yet scheduled).

ptarjan committed:
Hide update available when Tesla update is scheduled for later

When an update is scheduled far in the future (more than 2 minutes),
hide the update entirely by setting latest_version = installed_version.
This makes the state "off" so no update notification appears.

The update only shows as available when:
- Status is "available" (not yet scheduled)
- Status is "downloading" or "downloading_wifi_wait"
- Status is "installing"
- Status is "scheduled" AND within 2 minutes of starting

ptarjan committed:
Remove type ignores from Tesla update tests

ptarjan commented on homeassistant/components/tesla_fleet/update.py:
```diff
@@ -101,24 +105,24 @@ def _async_update_attrs(self) -> None:
-        return None
+    def _is_scheduled_soon(self) -> bool:
+        """Check if a scheduled update is within the threshold to start."""
+        scheduled_time_ms = self.get("vehicle_state_software_update_scheduled_time_ms")
```
Want me to search for more definitive documentation on the exact field name?

> yes

WebFetch
https://developer.tesla.com/docs/fleet-api/endpoints/vehicle-endpoints
WebFetch
https://tesla-api.timdorr.com/vehicle/state/softwareupdate
Request failed with status code 404
WebFetch
https://github.com/timdorr/tesla-api/blob/master/docs/vehicle/state/softwareupdate.md
Request failed with status code 404
WebFetch
https://tesla-api.timdorr.com/vehicle/commands/softwareupdate
Request failed with status code 503
WebSearch
Tesla API vehicle_data software_update scheduled_time_ms field response
The search confirms scheduled_time_ms is a valid field. According to the [Tesla JSON API (Unofficial)](https://tesla-api.timdorr.com/vehicle/commands/softwareupdate) documentation:

{
  "expected_duration_sec": 3000,
  "reason": "",
  "result": true,
  "scheduled_time_ms": 1685735308001,
  "status": "scheduled",
  "warning_time_remaining_ms": 120000
}

The field scheduled_time_ms is a Unix timestamp in milliseconds indicating when the update will install.

In Home Assistant's Tesla Fleet integration, the coordinator flattens nested JSON:

vehicle_state.software_update.scheduled_time_ms → vehicle_state_software_update_scheduled_time_ms
So the key I used should be correct.

Bre77 commented:
> Is this PR waiting on me for anything?

Just someone from core team to review and approve.
